examples/plotting/hmi_cutout.py

```diff
@@ -0,0 +1,126 @@
+"""
+====================
+HMI Showcase: Cutout
+====================
+
+This example demonstrates how to plot a cutout region of a `~sunpy.map.Map` with connector lines
+that indicate the region of interest in the full-disk image.
+
+Since this example deals with the creation of a specific style of image, there are multiple lines that deal directly with matplotlib axes.
+Furthermore, unlike other examples, since this deals with the creation of one figure, there are fewer breaks in the example.
+"""
+import matplotlib.colors
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.patches import ConnectionPatch
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.data.sample import HMI_LOS_IMAGE
+
+##############################################################################
+# Firstly, we will use the HMI LOS image within the sunpy sample data and focus
+# the cutout over an active region near the solar center.
+
+magnetogram = sunpy.map.Map(HMI_LOS_IMAGE).rotate()
+left_corner = SkyCoord(Tx=-142*u.arcsec, Ty=50*u.arcsec, frame=magnetogram.coordinate_frame)
+right_corner = SkyCoord(Tx=158*u.arcsec, Ty=350*u.arcsec, frame=magnetogram.coordinate_frame)
+
+##############################################################################
+# As there is "data" off the limb of the magetogram, we will need to mask it away.
+# We will create a second sunpy map that holds this masked data for use later.
+# From here the rest of the comments will be inside the code block.
+
+x, y = np.meshgrid(*[np.arange(v.value) for v in magnetogram.dimensions]) * u.pixel
+hpc_coords = magnetogram.pixel_to_world(x, y)
+solar_rad = np.sqrt(hpc_coords.Tx ** 2 + hpc_coords.Ty ** 2) / magnetogram.rsun_obs
+mask = np.ma.masked_greater(solar_rad, 1)
+magnetogram_big = sunpy.map.Map(magnetogram.data, magnetogram.meta, mask=mask.mask)
+
+
+# The next stage is to setup the figure in two stages.
+# The first stage is dealing with the full disk image.
+
+fig = plt.figure()
+
+# We want to create a nice normalation range for the image
+
+norm = matplotlib.colors.SymLogNorm(50, vmin=-7.5e2, vmax=7.5e2)
+
+# Plot the first magnetogram
+
+ax1 = fig.add_subplot(121, projection=magnetogram_big)
+magnetogram_big.plot(axes=ax1, cmap='RdBu_r', norm=norm, annotate=False,)
+
+# These lines deal with hiding the axis, its ticks and labels
+
+lon, lat = ax1.coords[0], ax1.coords[1]
```

Comment:
Maybe
```
for coord in ax1.coords:
    coord.frame.set_linewidth(0)
    ...
```
Would be nicer code? Maybe not though.

Reply:
Fixed.